lib/mix/lib/mix/tasks/xref.ex

```diff
@@ -191,4 +200,100 @@ defmodule Mix.Tasks.Xref do
   defp excluded?(module, func, arity, excludes) do
     MapSet.member?(excludes, module) or MapSet.member?(excludes, {module, func, arity})
   end
+
+  ## Callers
+
+  defp do_callers(filter) do
+    each_source_entries(&source_calls_for_filter(&1, filter), &print_calls/2)
+  end
+
+  defp source_calls_for_filter(source, filter) do
+    runtime_dispatches = source(source, :runtime_dispatches)
+    compile_dispatches = source(source, :compile_dispatches)
+    dispatches = merge_dispatches(runtime_dispatches, compile_dispatches)
```

Comment:
This could be improved by merging dispatches after filtering them, I suppose. Let me know if I should make the change.